src/Workspaces/SharedUtilitiesAndExtensions/Compiler/VisualBasic/Services/SyntaxFacts/VisualBasicSyntaxFacts.vb

```diff
@@ -2207,13 +2207,13 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.LanguageServices
                     Return DeclarationKind.Namespace
                 Case SyntaxKind.ImportsStatement
                     Return DeclarationKind.NamespaceImport
-                Case SyntaxKind.ClassBlock
+                Case SyntaxKind.ClassBlock, SyntaxKind.ClassStatement
                     Return DeclarationKind.Class
-                Case SyntaxKind.StructureBlock
+                Case SyntaxKind.StructureBlock, SyntaxKind.StructureStatement
                     Return DeclarationKind.Struct
-                Case SyntaxKind.InterfaceBlock
+                Case SyntaxKind.InterfaceBlock, SyntaxKind.InterfaceStatement
                     Return DeclarationKind.Interface
-                Case SyntaxKind.EnumBlock
+                Case SyntaxKind.EnumBlock, SyntaxKind.EnumStatement
```

Comment:
Few lines below, there is:

```
                Case SyntaxKind.SubStatement
                    If Not IsChildOf(declaration, SyntaxKind.SubBlock) Then
                        Return DeclarationKind.Method
                    End If
```

This is very weird, why the "IsChildOf" check is negated? and how `SubStatement` won't be a child of a `SubBlock`?

Reply:
This is intended. It's for cases like when you're on an interface and you only have the statement portion without a block